src/main.rs

```diff
@@ -254,3 +255,37 @@ fn CategoriesList(racers: Vec<Racer>, selected_category_id: Signal<Option<u32>>)
         }
     }
 }
+
+#[component]
+fn ResultInput() -> Element {
+    let mut start_number = use_signal(|| "".to_string());
+
+    rsx! {
+        form {
+             onsubmit: move |event| {
+                event.prevent_default();
+                println!("Enter pressed: {}", start_number.read());
+                // todo: store result
+                start_number.set(String::from(""));
+            },
+            input {
+                class: "form-control",
+                placeholder: "Start number",
+                type: "number",
+                value: start_number,
+                onkeydown: move |event| {
+                    let key = event.key().to_string();
+                    let allowed_specials = matches!(event.key(),Key::Enter);
```

Comment:
I think we should move bigger blocks into custom functions, as they are not linted :(

Reply:
Why `matches` is needed? Just comparison doesnt work? ..and maybe its `is_enter`?